### DISCUSSION TITLE
start_http_server port reuse

### DISCUSSION THREAD

**Discussion by KianaLia** [prometheus/client_python#809] [Q&A]
Hi guys!
I've asked a Q about reusing the **port** used in start_http_server(**port**). 
please guide me!
see: [https://stackoverflow.com/questions/72260685/updating-prometheus-metrics-http-server-48-address-already-in-use-error](url)

**Comment by csmarchbanks** [comment]
Would it be possible to call start_http_server from a single location rather than as part of the training process? Then the server will keep running on that port and the metrics will be updated as train is called.

**Reply by KianaLia** [reply]
Assume that I make a new main.py file which includes the start_http_server() command. How can I pass the metrics calculated in train.py's registry to be exposed in the server?
(note that a variable can't  be shared between python modules. the function calculating the variable should be called and the variable should be calculated every time)
